models/model_factory.py

```diff
@@ -0,0 +1,122 @@
+from torch import nn
+from models.resnet import resnet18
+from models.inception import inception3
+from dotenv import load_dotenv
+from utils.model_utils import split_model_file_name, get_model_info, AVAILABLE_MODELS
+from utils.time_utils import datetime_to_str, str_to_datetime
+import os
+from datetime import datetime
+import torch
+import pandas as pd
+
+load_dotenv()
+
+DATA_FOLDER = os.getenv("DATA_FOLDER_PATH")
+MODEL_FOLDER = os.path.join(DATA_FOLDER, "models")
+MODEL_DF = pd.read_csv(os.path.join(DATA_FOLDER, "models.csv"))
+
+def get_model_class(name: str, num_of_classes: int) -> nn.Module:
+
+  if name not in AVAILABLE_MODELS:
+    raise ValueError(f"Model type not supported, available models: {AVAILABLE_MODELS}")
+
+  # Names are defined in the class constructor function in the model desclarations
+  if name == 'resnet18':
+    return resnet18(num_classes=num_of_classes)
+  elif name == 'inception_v3':
+    return inception3(num_classes=num_of_classes)
+
+
+def get_trained_model_by_id(id: str) -> nn.Module:
+  models = os.listdir(MODEL_FOLDER)
+
+  # Filter models that don't contain the id
+  filtered_models = [model for model in models if id in models]
+
+  if len(filtered_models) == 0:
+    raise ValueError(f"Could not find model with id {id}")
+  elif len(filtered_models) > 0:
```

Comment:
```suggestion
  elif len(filtered_models) > 1:
```

Reply:
I think this should check if there are more than 1 model with the id. Otherwise this also catches the case where there is only 1 model with the correct id

Reply:
True